src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1208,82 +1122,126 @@ public static string GetPSModulePath(PSModulePathScope scope)
+#else
                 else
+#endif
                 {
-                    currentProcessModulePath += Path.PathSeparator;
+                    entry = entry.TrimEnd(dirSeparatorChars);
                 }
 
-                if (string.IsNullOrEmpty(hklmMachineModulePath)) // EVT.Machine does Not exist
+                // If after trimming we don't have a path value we ignore it.
+                if (entry.Length == 0)
                 {
-                    currentProcessModulePath += CombineSystemModulePaths(); // += (SharedModulePath + $PSHome\Modules)
+                    continue;
                 }
-                else
+
+                bool add = true;
+                foreach (ReadOnlySpan<char> listEntry in finalModulePath)
                 {
-                    currentProcessModulePath += hklmMachineModulePath; // += EVT.Machine
+                    if (listEntry.SequenceEqual(entry, valueComparer))
+                    {
+                        add = false;
+                        break;
+                    }
                 }
-            }
-            // EVT.Process exists
-            // Now handle the case where the environment variable is already set.
-            else
-            {
-                string personalModulePathToUse = string.IsNullOrEmpty(hkcuUserModulePath) ? personalModulePath : hkcuUserModulePath;
-                string systemModulePathToUse = string.IsNullOrEmpty(hklmMachineModulePath) ? psHomeModulePath : hklmMachineModulePath;
-
-                // Maintain order of the paths, but ahead of any existing paths:
-                // personalModulePath
-                // sharedModulePath
-                // systemModulePath
-
-                int insertIndex = 0;
 
-                currentProcessModulePath = UpdatePath(currentProcessModulePath, personalModulePathToUse, ref insertIndex);
-                currentProcessModulePath = UpdatePath(currentProcessModulePath, sharedModulePath, ref insertIndex);
-                currentProcessModulePath = UpdatePath(currentProcessModulePath, systemModulePathToUse, ref insertIndex);
-            }
-
-            return currentProcessModulePath;
-        }
-
-        private static string UpdatePath(string path, string pathToAdd, ref int insertIndex)
-        {
-            if (!string.IsNullOrEmpty(pathToAdd))
-            {
-                path = AddToPath(path, pathToAdd, insertIndex);
-                insertIndex = path.IndexOf(Path.PathSeparator, PathContainsSubstring(path, pathToAdd));
-                if (insertIndex != -1)
+                if (add)
                 {
-                    // advance past the path separator
-                    insertIndex++;
+#if UNIX
+                    string pathEntry = entry.ToString();
+#else
+                    string pathEntry;
+                    unsafe
+                    {
+                        // Weird code avoids multiple allocations to replace / with \.
```

Comment:
Can we use [MemoryExtensions.Replace](https://learn.microsoft.com/en-us/dotnet/api/system.memoryextensions.replace?view=net-8.0#system-memoryextensions-replace-1(system-readonlyspan((-0))-system-span((-0))-0-0)) in the string.Create?

Reply:
Looks like we should be able to, will try it out.

Reply:
Done